forage-android/src/main/java/com/joinforage/forage/android/ui/ForagePINEditText.kt

```diff
@@ -2,158 +2,88 @@ package com.joinforage.forage.android.ui
 
 import android.app.Application
 import android.content.Context
-import android.content.res.TypedArray
-import android.graphics.Color
-import android.text.InputType
+import android.graphics.Typeface
 import android.util.AttributeSet
-import android.util.TypedValue
 import android.widget.LinearLayout
-import com.google.android.material.textfield.TextInputLayout
+import com.basistheory.android.view.TextElement
 import com.joinforage.forage.android.LDManager
 import com.joinforage.forage.android.R
 import com.joinforage.forage.android.VaultConstants
-import com.joinforage.forage.android.network.ForageConstants
+import com.joinforage.forage.android.collect.BTPinCollector
+import com.joinforage.forage.android.collect.PinCollector
+import com.joinforage.forage.android.collect.VGSPinCollector
 import com.verygoodsecurity.vgscollect.widget.VGSEditText
-import com.verygoodsecurity.vgscollect.widget.VGSTextInputLayout
 
-/**
- * Material component using VGSTextInputLayout and VGSEditText to collect the PIN
- */
+interface PINEditText {
+    var isValid: Boolean
+    var isEmpty: Boolean
+    fun setTextColor(textColor: Int)
+    fun setTextSize(textSize: Float)
+    var typeface: Typeface?
+    fun setHint(hint: String)
+    fun setHintTextColor(hintTextColor: Int)
+}
+
 class ForagePINEditText @JvmOverloads constructor(
     context: Context,
     attrs: AttributeSet? = null,
     defStyleAttr: Int = R.attr.foragePanEditTextStyle
-) : LinearLayout(context, attrs, defStyleAttr) {
-    private val textInputLayout: VGSTextInputLayout
-    private val textInputEditText: VGSEditText
-    private var vaultType: String
+) : PINEditText, LinearLayout(context, attrs, defStyleAttr) {
+    private var vault: VaultWrapper?
 
     init {
         setWillNotDraw(false)
-
         orientation = VERTICAL
 
-        vaultType = LDManager.getVaultProvider(context.applicationContext as Application)
-        if (vaultType == VaultConstants.BT_VAULT_TYPE) {
-            // TODO: Do BT stuff!
-        } else if (vaultType == VaultConstants.VGS_VAULT_TYPE) {
-            // TODO: Do VGS stuff!
+        var vaultType = LDManager.getVaultProvider(context.applicationContext as Application)
+        vault = if (vaultType == VaultConstants.BT_VAULT_TYPE) {
+            BTVaultWrapper(context, attrs, defStyleAttr)
```

Comment:
This could be a problem:

I'll need to read the LDClient documentation first, but it takes time to initialize the client and fetch the feature flag. Here we are drawing the UI element; this might freeze the app / generate ANRs.

Probably we might need to create some init method inside `ForageSDK` to fetch this before we reach this part

Reply:
I have tested this locally and the launch darkly initialization code is executed prior to drawing the component. I haven't noticed any performance issues or app freezing (however it's very possible that this could show up once the code is used on an actual Android device and not an emulator). From a first glance this strategy seems to be working, but if you have more insight please share!